### DISCUSSION TITLE
Bun Run Script in All Workspace Packages

### DISCUSSION THREAD

**Discussion by alexkarpovnow** [oven-sh/bun#9385] [Q&A]
Does bun have a way to run a script, e.g. 'bun run build' in every package.json for every workspace? Cerca 'npx lerna exec'

**Comment by spion** [comment]
I've forked `wsrun` into `wsrun-ng` and added support for detecting npm_execpath, which makes it automatically work with `bun`: https://github.com/spion/wsrun-ng

**Comment by alexeyten** [comment]
Does this works?
```
bun run --filter '*' build
```

https://bun.sh/docs/cli/filter

**Comment by spion** [comment]
That should work, but it doesn't seem to fully respect dependency order in https://github.com/ionic-solidjs/ionic-solidjs

**Comment by skoenfaelt** [comment]
Dear Reader from the Web,
you might face the same issue like me.

> "I use --filter but it wont work"

Its important to place `--filter '*'` BEFORE the actual script.

DON´T
```bash
bun run dev:prepare --filter 'layers/*'
```

DO
```bash
bun run --filter 'layers/*' dev:prepare 
```

**Comment by PaulRBerg** [comment]
Bun should offer a nicer syntax for running a script in all workspaces. This syntax:

```
bun run --filter '*'
```

Is so much uglier/less clear compared to Yarn Berry:

```
yarn workspaces foreach run
```
